### PR TITLE
Numpy literals

### PR DESCRIPTION
**Pull request by johnc1231**
This PR adds the ability to use `hl.literal` on numpy arrays. It also changes the behavior of `hl._ndarray` when called on a numpy ndarray so it just goes through literal to avoid creating an IR node for each element of tensor.

### CODE REVIEW

**Comment by johnc1231** [comment]
Partially addresses #7122

File: hail/python/hail/expr/types.py [hunk 1]
```diff
@@ -1647,6 +1656,16 @@ def types_match(left, right) -> bool:
...
 
+def from_numpy(np_dtype):
+    if np_dtype == np.int32:
+        return tint32
+    elif np_dtype == np.int64:
+        return tint64
+    elif np_dtype == np.float32:
+        return tfloat32
+    elif np_dtype == np.float64:
+        return tfloat64
```
Comment (tpoterba) [inline hail/python/hail/expr/types.py]: needs a base case that errors

**Comment by johnc1231** [comment]
Lack of test was me being lazy since I moved all the tests in my other ndarray PR. I'll just add one here and address the merge conflict in either direction.

**Comment by johnc1231** [comment]
Agh, mostly unrelated tests are failing because of a bug that I've fixed in a different PR. I'll wait for that to go in (should go in tomorrow), then rebase and dismiss

**Comment by johnc1231** [comment]
Actually closing this for a minute. Realized I'm not respecting row / column major semantics.

**Comment by johnc1231** [comment]
Reopening this. I've addressed your comments, and also made the decision to take the `row_major` argument off of `_ndarray`, since it adds unnecessary complexity that our users don't benefit from since we are reading everything into json anyway.

File: hail/src/main/scala/is/hail/expr/types/encoded/EBaseStruct.scala [hunk 2]
```diff
@@ -86,6 +86,9 @@ final case class EBaseStruct(fields: IndexedSeq[EField], override val required:
         PTupleField(idx, pt)
       }
       PTuple(pFields, required)
+    case t: TNDArray =>
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/expr/types/encoded/EBaseStruct.scala]: I'm a little unsure about this stuff, but good for now.

**Pull request merged**